src/gui/widgets/text_box_base.cpp

```diff
@@ -630,7 +660,11 @@ void text_box_base::signal_handler_sdl_key_down(const event::ui_event event,
 
 		case SDLK_RETURN:
 		case SDLK_KP_ENTER:
-			if(!is_composing() || (modifier & (KMOD_CTRL | KMOD_ALT | KMOD_GUI | KMOD_SHIFT))) {
+			if ((modifier & KMOD_SHIFT) && is_editable()) {
+				insert_char("\n");
```

Comment:
This doesn't quite seem right either. It's correct for multiline text, but is this going to insert a newline into a single-line text field too?

Reply:
Will try to handle this in multiline_text itself, instead of text_box_base.

Reply:
Handling moved to multiline text itself. Pressing Enter with any modifiers causes a linebreak to be inserted.